### PR TITLE
refactor: consolidate jobs in autograding.yml to improve execution speed

### LINKED ISSUE #3: [Feat]: speed up autograding with caching
Look into [this](https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies) to speed up the autograding workflow.

### PR DESCRIPTION
**Pull request by David-GERARD**
Doing this to avoid going over the 1000min/month limit that the free tier grants organisations.

it makes the run go from 3min to 20s

### CODE REVIEW

File: .github/workflows/autograding.yml [hunk 1]
```diff
@@ -20,54 +20,25 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version-file: 'pyproject.toml' # Read python version from a file pyproject.toml
+        cache: 'pip'
```
Comment (mikicz) [inline .github/workflows/autograding.yml:23]: I think it should be `poetry` if we are using pyproject.toml, no?

File: .github/workflows/autograding.yml [hunk 2]
```diff
@@ -20,54 +20,25 @@ jobs:
...
         
-    # Check Task 2: Poetry Setup
-    - name: Check Task 2
-      run: |
         if [ -f "pyproject.toml" ] && [ -f "poetry.lock" ]; then
           echo "Poetry setup verified."
         else
           echo "Error: Poetry files not found."
           exit 1
         fi
```
Comment (mikicz) [inline .github/workflows/autograding.yml]: I think if these two files don't exist then `poetry install` might not work?

**Review by mikicz — APPROVED** [review]

File: .github/workflows/autograding.yml (hunk 1 above)
Comment (David-GERARD) [reply .github/workflows/autograding.yml:23]: changing 'pip' with 'poetry' result in the following error message: `Error: Unable to locate executable file: poetry. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.`

Using the poetry action probable would solve this, but I leave it for a later re-work of the code.

File: .github/workflows/autograding.yml (hunk 2 above)
Comment (David-GERARD) [reply .github/workflows/autograding.yml]: Yep that's true, I'll refactor.

**Pull request merged**